### PR TITLE
Fix .NET Native queries with shadow properties

### LINKED ISSUE #3604: Queries with shadow properties fail on .NET Native
Any query that uses `EF.Property(e, "SomeProperty")` fails on UWP, .NET Native with this error:

`System.InvalidOperationException: The EF.Property<T> method may only be used within LINQ queries.`

### PR DESCRIPTION
**Pull request by natemcmaster**
Depends on #4320 

Fix #3604

cc @anpete

### CODE REVIEW

File: src/EntityFramework.Core/Query/EntityQueryModelVisitor.cs [hunk 1]
```diff
@@ -32,9 +32,22 @@ public abstract class EntityQueryModelVisitor : QueryModelVisitorBase
...
-        public static readonly MethodInfo PropertyMethodInfo
+        private static readonly MethodInfo PropertyMethodInfo
             = typeof(EF).GetTypeInfo().GetDeclaredMethod(nameof(EF.Property));
 
+        private const string EFTypeName = "Microsoft.Data.Entity.EF";
+
+        public static bool IsPropertyMethod([CanBeNull] MethodInfo methodInfo) =>
+            methodInfo != null
+            && methodInfo.IsGenericMethod
+#if NETCORE50
```
Comment (anpete) [inline src/EntityFramework.Core/Query/EntityQueryModelVisitor.cs]: I would be OK with just the NETCORE50 approach if we wanted to simplify further.

File: src/EntityFramework.Core/Query/EntityQueryModelVisitor.cs [hunk 2]
```diff
@@ -32,9 +32,22 @@ public abstract class EntityQueryModelVisitor : QueryModelVisitorBase
         public static readonly ParameterExpression QueryContextParameter
             = Expression.Parameter(typeof(QueryContext), "queryContext");
 
-        public static readonly MethodInfo PropertyMethodInfo
+        private static readonly MethodInfo PropertyMethodInfo
             = typeof(EF).GetTypeInfo().GetDeclaredMethod(nameof(EF.Property));
 
+        private const string EFTypeName = "Microsoft.Data.Entity.EF";
```
Comment (anpete) [inline src/EntityFramework.Core/Query/EntityQueryModelVisitor.cs]: Consider typeof(EF).FullName

**Comment by anpete** [comment]
:shipit:

File: src/EntityFramework.Core/Query/EntityQueryModelVisitor.cs (hunk 1 above)
Comment (natemcmaster) [inline src/EntityFramework.Core/Query/EntityQueryModelVisitor.cs]: So, not worried about the perf difference between string comparison and reference comparisons?

Comment (anpete) [inline src/EntityFramework.Core/Query/EntityQueryModelVisitor.cs]: No, it should be negligible in the context of query compilation.

**Pull request merged**